backend/users/models.py

```diff
@@ -23,3 +89,106 @@ class User(AbstractBaseUser, PermissionsMixin):
 
     def get_short_name(self):
         return self.email
+
+    def clean(self):
+        # TODO check required field here i.e. company-required-fileds?
+        pass
+
+    def clean_business_fields(self):
+
+        if not self.business_name:
+            raise exceptions.ValidationError(
+                {"business_name": _("Missing Business Name in your user profile.")}
+            )
+
+        if not self.phone_number:
+            raise exceptions.ValidationError(
+                {"phone_number": _("Missing Phone Number in your user profile.")}
+            )
+
+        if not self.country:
+            raise exceptions.ValidationError(
+                {"country": _("Missing Country in your user profile.")}
+            )
+
+        if not self.is_eu():
+            return
+
+        if self.is_italian():
+
+            # TODO check fiscal code with italian roles...
+            #  but first you have to split the address for get CAP!
+
+            if not self.fiscal_code and not self.vat_number:
+                raise exceptions.ValidationError(
+                    _("Please specify Fiscal Code or VAT number in your user profile.")
+                )
+
+            if self.fiscal_code:
+                self.clean_fiscal_code()
+
+            if not self.recipient_code and not self.pec_address:
+                raise exceptions.ValidationError(
+                    _(
+                        "For Italian companies for electronic invoicing it is "
+                        "mandatory to specify the recipient's code or the pec address."
+                    )
+                )
+            return
+
+        if not self.vat_number:
+            raise exceptions.ValidationError(
+                {"vat_number": _("Missing VAT Number in your user profile.")}
+            )
+
+    def clean_fiscal_code(self):
+        msg = _("Fiscal Code is invalid")
+        if not len(self.fiscal_code) == 16:
```

Comment:
I am not a super expert of the italian fiscal code, but if we use the User model for companies as well the length of the code might be 11

Reply:
Fiscal Code is different from the VAT maybe you are confusing the two fields? 
Fiscal Code is only for people or freelancer I think...

Reply:
This method is specific for italian fiscal code ("SSSNNNYYMDDZZZZX"). If there are some roles for VAT or companies code  I have to develop another method to check that code... 
Maybe I can put a docstring with these notes...

Reply:
As we decided to remove any company field, for now, I resolve this.